### PR TITLE
[wasm] Enable tests blocked by IL2105

### PR DESCRIPTION
**Pull request by lewing**

### CODE REVIEW

**Comment by lewing** [comment]
resolves https://github.com/dotnet/runtime/issues/51602

**Comment by lewing** [comment]
I skipped the suppression so there might be some errors still.

**Comment by lewing** [comment]
a stack of https://github.com/dotnet/core-eng/issues/13691 here :(

**Comment by lewing** [comment]
```
@azure-pipelinesazure-pipelines
/ runtime (Libraries Build windows allConfigurations x64 Debug)
.dotnet\sdk\6.0.100-rc.1.21411.28\NuGet.RestoreEx.targets#L19
.dotnet\sdk\6.0.100-rc.1.21411.28\NuGet.RestoreEx.targets(19,5): error : (NETCORE_ENGINEERING_TELEMETRY=Restore) Failed to retrieve information about 'NETStandard.Library.Ref' from remote source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/c9f8ac11-6bd8-4926-8306-f075241547f7/nuget/v3/flat2/netstandard.library.ref/index.json'.
   at NuGet.Protocol.HttpFileSystemBasedFindPackageByIdResource.FindPackagesByIdAsync(String id, SourceCacheContext cacheContext, ILogger logger, CancellationToken cancellationToken)
   at NuGet.Protocol.HttpFileSystemBasedFindPackageByIdResource.EnsurePackagesAsync(String id, SourceCacheContext cacheContext, ILogger logger, CancellationToken cancellationToken)
   at NuGet.Protocol.HttpFileSystemBasedFindPackageByIdResource.DoesPackageExistAsync(String id, NuGetVersion version, SourceCacheContext cacheContext, ILogger logger, CancellationToken cancellationToken)
   at NuGet.Commands.SourceRepositoryDependencyProvider.FindLibraryCoreAsync(LibraryRange libraryRange, SourceCacheContext cacheContext, ILogger logger, CancellationToken cancellationToken)
   at NuGet.Commands.SourceRepositoryDependencyProvider.<>c__DisplayClass19_0.<<FindLibraryAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at NuGet.Commands.SourceRepositoryDependencyProvider.FindLibraryAsync(LibraryRange libraryRange, NuGetFramework targetFramework, SourceCacheContext cacheContext, ILogger logger, CancellationToken cancellationToken)
   at NuGet.DependencyResolver.ResolverUtility.<>c__DisplayClass9_1.<<FindLibraryFromSourcesAsync>b__0>d.MoveNext()
```
is one I haven't seen before

**Comment by lewing** [comment]
cc @danmoseley

File: eng/testing/tests.mobile.targets [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <!-- Suppress linker warnings as these are tests -->
     <SuppressTrimAnalysisWarnings>true</SuppressTrimAnalysisWarnings>
     <!-- Can be removed when SDK update picks up https://github.com/dotnet/sdk/pull/18655. -->
```
Comment (stephentoub) [inline eng/testing/tests.mobile.targets:16]: This PR was merged two weeks ago.  Is the comment still accurate?

Comment (lewing) [reply eng/testing/tests.mobile.targets]: No, the current plan is to continue to suppress warnings on this leg.  I removed the comment.

**Review by akoeplinger — APPROVED** [review]

**Comment by lewing** [comment]
this only turns on new tests for the browser lanes and they are green.

**Pull request merged**